modules/core/04-channel/types/msgs.go

```diff
@@ -589,16 +589,59 @@ var _ sdk.Msg = &MsgChannelUpgradeCancel{}
 
 // NewMsgChannelUpgradeCancel constructs a new MsgChannelUpgradeCancel
 // nolint:interfacer
-func NewMsgChannelUpgradeCancel() *MsgChannelUpgradeCancel {
-	return &MsgChannelUpgradeCancel{}
+func NewMsgChannelUpgradeCancel(
+	portID, channelID string,
+	errorReceipt ErrorReceipt,
+	proofErrReceipt []byte,
+	proofHeight clienttypes.Height,
+	signer string,
+) *MsgChannelUpgradeCancel {
+	return &MsgChannelUpgradeCancel{
+		PortId:            portID,
+		ChannelId:         channelID,
+		ErrorReceipt:      errorReceipt,
+		ProofErrorReceipt: proofErrReceipt,
+		ProofHeight:       proofHeight,
+		Signer:            signer,
+	}
 }
 
 // ValidateBasic implements sdk.Msg
 func (msg MsgChannelUpgradeCancel) ValidateBasic() error {
-	return errors.New("error method not implemented")
+	if err := host.PortIdentifierValidator(msg.PortId); err != nil {
+		return sdkerrors.Wrap(err, "invalid port ID")
+	}
+
+	if !IsValidChannelID(msg.ChannelId) {
+		return ErrInvalidChannelIdentifier
+	}
+
+	if len(msg.ProofErrorReceipt) == 0 {
+		return sdkerrors.Wrap(commitmenttypes.ErrInvalidProof, "cannot submit an empty proof")
+	}
+
+	if msg.ProofHeight.IsZero() {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidHeight, "proof height must be non-zero")
+	}
+
+	if msg.ErrorReceipt.Sequence == 0 {
```

Comment:
[q] maybe this isn't possible, but should this be `<= 0`, a negative value is currently valid. The same applies for `ProofErrorReceipt`

Reply:
The sequence field is a unsigned int as far as I know

Reply:
ah perfect, not an issue then 👍